lib/src/model/relation/relation_repository_providers.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:logging/logging.dart';
+
+import 'package:lichess_mobile/src/model/auth/auth_client.dart';
+import 'package:lichess_mobile/src/model/user/user.dart';
+import 'relation_repository.dart';
+
+part 'relation_repository_providers.g.dart';
+
+@Riverpod(keepAlive: true)
+RelationRepository relationRepository(RelationRepositoryRef ref) {
+  final apiClient = ref.watch(authClientProvider);
+  return RelationRepository(
+    logger: Logger('RelationRepository'),
+    apiClient: apiClient,
+  );
+}
+
+@riverpod
```

Comment:
we should cache this with `@Riverpod(keepAlive: true)`

Reply:
with @Riverpod(keepAlive: true) I have this behavior, steps to reproduce:

- go to following screen
- unfollow user
- go back to friends screen
- go to following screen

it serves old results, not updated with unfollow request at step 2
because cached with @Riverpod(keepAlive: true)

Reply:
yes I see... the unfollow action should also invalidate the cache with `ref.invalidate`

Reply:
Now I've added @Riverpod(keepAlive: true) and ref.invalidate, but currently, the UI list updates only after the unfollow request is completed

Reply:
Now I've added `@Riverpod(keepAlive: true)` and `ref.invalidate` :)